hack/generated/pkg/genruntime/arm_transformer.go

```diff
@@ -15,7 +15,7 @@ type ToARMConverter interface {
 }
 
 type FromARMConverter interface {
-	CreateEmptyARMValue() interface{}
+	CreateEmptyARMValue() ARMResourceStatus
```

Comment:
This type being an `XStatus` type suggests that this interface won't work for fields in the `Spec` graph, is that right? We'll need that eventually I think for diffing. IIRC `ARMResourceStatus` was basically an empty interface though so possibly we can just rename it to something and allow it to apply everywhere when we get to that point?

Just wondering if you've thought any about that.

Reply:
It's on my radar. 😀 

In my WIP branch I've got three different sets of types - one for Statusess (those directly referenced by resources), one for Specs  (ditto), and a generic set for other types. 

I suspected we'd need them on Spec types for diff'ing, thanks for confirming.

Given the three sets noted above, do we need `CreateEmptyARMValue()` on the other types? Looks (from what I've seen so far) as though we only need it as an entry point at the top of a structure.